flink-runtime/src/test/java/org/apache/flink/runtime/state/TaskStateManagerImplTest.java

```diff
@@ -97,6 +104,39 @@ public void testStateReportingAndRetrieving() {
 		Assert.assertNull(taskStateManager.operatorStates(operatorID_3));
 	}
 
+	/**
+	 * This tests if the {@link TaskStateManager} properly returns the the subtask local state dir from the
+	 * corresponding {@link TaskLocalStateStore}.
+	 */
+	@Test
+	public void testForwardingSubtaskLocalStateBaseDirFromLocalStateStore() throws IOException {
+		JobID jobID = new JobID(42L, 43L);
+		JobVertexID jobVertexID = new JobVertexID(12L, 34L);
+		ExecutionAttemptID executionAttemptID = new ExecutionAttemptID(23L, 24L);
+		TestCheckpointResponder checkpointResponderMock = new TestCheckpointResponder();
+
+		TemporaryFolder tmpFolder = new TemporaryFolder();
+
+		try {
+			tmpFolder.create();
+			TaskLocalStateStore taskLocalStateStore =
+				new TaskLocalStateStore(jobID, jobVertexID, 13, tmpFolder.newFolder());
+
+			TaskStateManager taskStateManager = taskStateManager(
+				jobID,
+				executionAttemptID,
+				checkpointResponderMock,
+				null,
+				taskLocalStateStore);
+
+			Assert.assertEquals(
+				taskLocalStateStore.getSubtaskLocalStateBaseDirectory(),
+				taskStateManager.getSubtaskLocalStateBaseDirectory());
```

Comment:
Why does the `TaskStateManager` knows about the subtask local state base directory?

Reply:
Already changed by later commits.